hipify-clang/CMakeLists.txt

```diff
@@ -80,7 +80,21 @@ endif()
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${EXTRA_CFLAGS}")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${EXTRA_CFLAGS} ${StdCpp} -DHIPIFY_CLANG_RES=\\\"${LLVM_LIBRARY_DIRS}/clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}\\\"")
 
-install(TARGETS hipify-clang DESTINATION bin)
+install(TARGETS hipify-clang DESTINATION ${CMAKE_INSTALL_PREFIX})
+
+install(
+  DIRECTORY ${LLVM_DIR}/../../clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}/
```

Comment:
I think if LLVM_DIR is pointing to /opt/rocm/llvm, then this would choose directory /opt/clang/major.minor.patch. Is that the behaviour we want?

Reply:
`${LLVM_DIR}` is calculated by cmake based on `CMAKE_PREFIX_PATH`.
If `CMAKE_PREFIX_PATH` points to LLVM dist root dir, then `LLVM_DIR` points to `<LLVM dist>/lib/cmake/llvm`.

Here is an output example:
```
emankov@spb-hlc-fiji1h:~/git/HIP/hipify-clang/build$ cmake -DCMAKE_PREFIX_PATH=/home/emankov/git/LLVM/9.0.0/llvm_x64_release/dist -DHIPIFY_CLANG_TESTS=1 -DCUDA_TOOLKIT_ROOT_DIR=/usr/local/cuda-10.1 -DCUDA_CUB_ROOT_DIR=/home/emankov/git/cub  -DCUDA_DNN_ROOT_DIR=/home/emankov/cudnn-10.1-v7.5.0.56  -DLLVM_EXTERNAL_LIT=/home/emankov/git/LLVM/9.0.0/llvm_x64_release/bin/llvm-lit -DCMAKE_BUILD_TYPE=Release ../ -DCMAKE_INSTALL_PREFIX=/home/emankov/git/HIP/hipify-clang/dist
-- Found LLVM 9.0.0:
--    - CMake module path: /home/emankov/git/LLVM/9.0.0/llvm_x64_release/dist/lib/cmake/llvm
--    - Include path     : /home/emankov/git/LLVM/9.0.0/llvm_x64_release/dist/include
--    - Binary path      : /home/emankov/git/LLVM/9.0.0/llvm_x64_release/dist/bin
-- Linker detection: GNU ld
-- Found lit: /usr/local/bin/lit
-- Found FileCheck: /home/emankov/git/LLVM/9.0.0/llvm_x64_release/dist/bin/FileCheck
-- Configuring done
-- Generating done
-- Build files have been written to: /home/emankov/git/HIP/hipify-clang/build
emankov@spb-hlc-fiji1h:~/git/HIP/hipify-clang/build$ make install
[100%] Built target hipify-clang
Install the project...
-- Install configuration: "Release"
-- Installing: /home/emankov/git/HIP/hipify-clang/dist/hipify-clang
-- Up-to-date: /home/emankov/git/HIP/hipify-clang/dist
-- Installing: /home/emankov/git/HIP/hipify-clang/dist/include
-- Installing: /home/emankov/git/HIP/hipify-clang/dist/include/__clang_cuda_device_functions.h
-- Installing: /home/emankov/git/HIP/hipify-clang/dist/include/avxintrin.h
-- Installing: /home/emankov/git/HIP/hipify-clang/dist/include/avx2intrin.h
-- Installing: /home/emankov/git/HIP/hipify-clang/dist/include/clflushoptintrin.h
-- Installing: /home/emankov/git/HIP/hipify-clang/dist/include/enqcmdintrin.h
...
```
where `CMAKE_PREFIX_PATH=/home/emankov/git/LLVM/9.0.0/llvm_x64_release/dist`
and  Found CMake module path (`LLVM_DIR`): `/home/emankov/git/LLVM/9.0.0/llvm_x64_release/dist/lib/cmake/llvm`
The direcory choosen as SORCE for install is: `/home/emankov/git/LLVM/9.0.0/llvm_x64_release/dist/lib/clang/9.0.0/`

So **all is needed is to set CMAKE_PREFIX_PATH correctly**.

Reply:
Oh I didn't know it pointed to the cmake directory. This makes sense now. Thanks